extensions/hooks/copy_pdbs_to_package.py

```diff
@@ -0,0 +1,30 @@
+from conan.tools.files import copy
+import glob
+import json
+import os
+import re
+
+
+def post_package(conanfile):
+    # Find dumpbin path
+    conanfile.run(
+        r'"%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe" -find "**\dumpbin.exe" -format json > out.json')
```

Comment:
If the ``out.json`` is generated inside the ``package_folder`` (likely), it will dirty the package. Must be saved somewhere else or made sure to be deleted.

Reply:
What happens when this ``run()`` executes and vswhere not found? If this hook is installed for different compilers (not msvc) in Windows?

Reply:
> What happens when this run() executes and vswhere not found? If this hook is installed for different compilers (not msvc) in Windows?

We should check that is running in Windows and using msvc. The hook is exclusively meant for debugging with Visual.

Reply:
`conanfile.run()` lets you redirect the stdout and stderr outputs, see https://github.com/jcar87/conan/blob/ab4895dae086bf4b6b45101546b3b8ebd9f794e4/conans/model/conan_file.py#L322 - this way we avoid writing a file anywhere

```
from io import StringIO

        output = StringIO()
        self.run("cmake --version", stdout=output)
        output_str = str(output.getvalue())
```
(I think it would work roughly like this)


the hook also has access to conanfile, so it can check if we are on Windows/MSVC (conanfile.settings)